src/librustdoc/html/static/main.js

```diff
@@ -145,6 +146,17 @@
             for (i = from; i <= to; ++i) {
                 addClass(document.getElementById(i), 'line-highlighted');
             }
+        } else if (ev !== null && search && !hasClass(search, "hidden") && ev.newURL) {
+            addClass(search, "hidden");
+            removeClass(document.getElementById("main"), "hidden");
+            var hash = ev.newURL.slice(ev.newURL.indexOf('#') + 1);
+            if (browserSupportsHistoryApi()) {
+                history.replaceState(hash, "std - Rust", "?search=#" + hash);
```

Comment:
* Firefox ignores the title argument to `pushState`/`replaceState` anyway, so right now this is moot, but:
* Setting this unilaterally to "std - Rust" is wildly inaccurate, even within libstd, because for anything but the crate root the title is set to the path of the item. But for anything *but* libstd, this is just erroneous.

On the other hand, with the empty search param, the link is now stable, and the search box and results are still the same. If we can't change the title (in Firefox, at least) via `replaceState`, then i'll just give up on it for the time being. I'd suggest either finding some other way to set it properly or just changing this back to `""`.